src/Dictionaries/getDictionaryConfigurationFromAST.cpp

```diff
@@ -166,6 +166,13 @@ void buildLayoutConfiguration(
                 value_field.getTypeName());
         }
 
+        if (value_field.safeGet<UInt64>() == 0)
```

Comment:
Did you test what happens if a negative value is passed as Float or String (--> l. 161), e.g. `-2.0`? I could imagine that `safeGet<UInt64>` does a numeric roll-over internally and we'll get UINT_MAX - 1 in which case the new check would not work.

Please also add a small SQL test --> https://clickhouse.com/docs/development/tests in tests/queries/0_stateless/ - thanks!

Reply:
yes, I tested with negative value and it fails in the previous check. It detects the negative value as `Int64`, then it throws an error.

Reply:
added the test with cc40fc3c996d4eb09f93776d6ef02c435cc78b93